File: zig/src/algorithms/bubble_sort.zig
const std = @import("std");

pub fn bubbleSort(array: []i32) void {
    var i: usize = 0;
    var j: usize = 0;

    while (i < array.len) : (i += 1) {
        while (j < array.len - 1 - i) : (j += 1) {
            if (array[j]  > array[j + 1]) {
                const tmp = array[j];
                array[j] = array[j + 1];
                array[j + 1] = tmp;
            }
        }
    }

    std.debug.print("{d}\n", .{array});
}

File: rust/src/algorithms/bubble_sort.rs
fn bubble_sort(array: &mut [i32]) {
    for i in 0..array.len() {
        for j in 0..array.len() - 1 - i {
            if array[j] > array[j + 1] {
                array.swap(j, j + 1);
            }
        }
    }
}

#[cfg(test)]
mod tests {
    use super::*;


    #[test]
    fn test_bubble_sort_1() {
        let mut array = [5, 2, 9, 1, 5, 6];
        bubble_sort(&mut array);
        assert!(array.is_sorted());
    }

    #[test]
    fn test_bubble_sort_2() {
        let mut array = [10, 6, 90, 2, 43, 56, 9];
        bubble_sort(&mut array);
        assert!(array.is_sorted());
    }
}
